Assorted documentation

derekdreery opened the pull request:
I'm just trying to learn how this lib works, and I'm looking at source to work it out. Hopefully if I doc as I go, others that come after me will benefit :)

jdm commented on markup5ever/serialize.rs:
```diff
@@ -6,35 +6,64 @@
 
 //§ serializing-html-fragments
+/// Used as a parameter to `serialize`, telling it if we want to skip the parent.
+/// TODO check: Currently the parameter to `ChildrenOnly` is unused.
```
It's used by the HTML serializer code: https://github.com/derekdreery/html5ever/blob/30891d283147bfa815d1ae47694ff97761b2f150/html5ever/src/serialize/mod.rs#L77

jdm commented on markup5ever/serialize.rs:
```diff
@@ -6,35 +6,64 @@
 pub enum TraversalScope {
+    /// Include the parent node when serializing.
     IncludeNode,
+    /// Only serialize the children of the node.
```
`/// Only serialize the children of the node, treating any provided qualified name as the parent while serializing.`

jdm commented on markup5ever/util/buffer_queue.rs:
```diff
@@ -123,6 +176,29 @@ impl BufferQueue {
     // If so, consume them and return Some(true).
     // If they do not match, return Some(false).
     // If not enough characters are available to know, return None.
+    /// Consume bytes matching the pattern, using a custom comparison function `eq`.
```
Let's remove the existing comments, since the new ones are more thorough.

derekdreery commented:
I've addressed the issues you've raised. You're welcome to merge this, or if you leave it open I can continue adding documentation to it.

Or if you would rather I'll start a new PR with any other documentation I do. I don't mind either way.

jdm commented:
Let's merge the changes that exist right now so others can benefit.
@bors-servo: r+

bors-servo commented:
:pushpin: Commit de94da4 has been approved by `jdm`

<!-- @bors-servo r=jdm de94da4a46c0d1bb844d0da323906707bf6a0cbb -->

bors-servo commented:
:hourglass: Testing commit de94da4a46c0d1bb844d0da323906707bf6a0cbb with merge 77a52cab6654462d2e8c177cd8231d9cf0e9f414...

bors-servo commented:
:sunny: Test successful - [status-travis](https://travis-ci.org/servo/html5ever/builds/304906251?utm_source=github_status&utm_medium=notification)
Approved by: jdm
Pushing 77a52cab6654462d2e8c177cd8231d9cf0e9f414 to master...
